lib/service/service.go

```diff
@@ -294,6 +294,10 @@ type TeleportProcess struct {
 	// importedDescriptors is a list of imported file descriptors
 	// passed by the parent process
 	importedDescriptors []FileDescriptor
+	// listenersBlocked is a flag that indicates that the process should not open
+	// new listeners (for instance, because we're shutting down and we've already
+	// closed all the listeners)
+	listenersBlocked bool
```

Comment:
nit: `listenersBlocked` naming suggest that listener can be also unblocked. 
Can we rename the flag to shutdown and set  in `StartShutdown`  function ?

Reply:
How about `listenersClosed`? I wanted this to be listener-specific as we might want to not necessarily tie the closing of listeners to the shutdown procedure in the future.